runtime/src/accounts_db.rs

```diff
@@ -6728,38 +6728,75 @@ pub mod tests {
     #[test]
     fn test_storage_remove_account_with_reset_then_access() {
```

Comment:
@carllin , @sakridge 
I will need help getting this test to do the right thing. There is still too much I don't understand about caching, clean, etc. Stephen's goal was to test:
a case where since we don't reset the accounts, that we might look at the store and it was reset before, but now it has data.

In this scenario, I can't get the remove_account call to get hit from the clean (or shrink). I apologize - this code is really messy. I have been trying a lot of different approaches and looking at 'similar' tests. I think we shouldn't hold up this PR further for me to learn all I need to learn to get this tested right. We can talk Monday.

Reply:
I can confirm the test is now hitting the code path in remove_account for the clean. I am not sure which shrink @sakridge meant with this slack:
"we need a test that does a shrink then tests that it can still access the account data, and also for a clean" I'll try other variations of shrink tomorrow. shrink_all_slots isn't currently hitting remove_account as far as I can tell.

Reply:
I merged changes from carl's test and mine. I now confirm hitting the remove_account case with both shrink and clean.